sched/environ/env_dup.c

```diff
@@ -61,82 +61,98 @@
  *
  ****************************************************************************/
 
-int env_dup(FAR struct task_group_s *group)
+int env_dup(FAR struct task_group_s *group, FAR char * const *envp_in)
 {
-  FAR struct tcb_s *ptcb = this_task();
   FAR char **envp = NULL;
   size_t envc = 0;
   int ret = OK;
 
-  DEBUGASSERT(group != NULL && ptcb != NULL && ptcb->group != NULL);
+  DEBUGASSERT(group != NULL);
 
   /* Pre-emption must be disabled throughout the following because the
    * environment may be shared.
    */
 
   sched_lock();
 
-  /* Does the parent task have an environment? */
+  /* Is there an environment ? */
 
-  if (ptcb->group != NULL && ptcb->group->tg_envp != NULL)
+  if (envp_in == NULL)
```

Comment:
Can you modify as below:
```
  if (envp_in || (envp_in = ptcb->group->tg_envp))
    {
      ... 
    }
```
to reduce the change.

Reply:
Sure